NumberTheory/ExtendedEuclid.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+    Extended Euclidean Algorithm
+    ----------------------------
+
+    Given two non-negative numbers a & b as inputs, find M[0], M[1] such that
+
+    gcd(a, b) = M[0]*a + M[1]*b
+
+    This algorithm has a variety of uses - in particular, it can be used to
+    quickly compute multiplicative inverses in finite fields, as is often
+    needed for cryptographic algorithms
+
+    Time Complexity
+    ---------------
+    O(log10(K)), where K = min(A, B) is the smaller of the input integers A and B
+
+    Space Complexity
+    ----------------
+    O(1)   
+*/
+
+#include<iostream>
+#include<vector>
+
+using namespace std;
+
+/*
+
+    extended_euclidean
+    --------------
+    Uses the recurrence relation,
+        GCD(a, b) = GCD(b, a%b),
+    to find coefficients M[0] and M[1] such that 
+        GCD(a, b) = M[0]*a + M[1]*b
+
+    Return value
+    ------------
+    vector<int> M of length 2 which satisfies the equation above
+
```

Comment:
Since there will always be exactly 2 values that the function would return, it would be better to return them in either a `tuple<int, int>` or an `array<int, 2>`.